### PR TITLE
CIV-7380 CivilDocumentStitchingService: unit-tests

### PR DESCRIPTION
**Pull request by UshaPanneerselvam1**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/CIV-7380

### Change description ###
CivilDocumentStitchingService: unit-tests and code cleanup

**Does this PR introduce a breaking change?** (check one with "x")
```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/test/java/uk/gov/hmcts/reform/civil/service/stitching/CivilDocumentStitchingServiceTest.java [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+            .createdDatetime(LocalDateTime.now())
+            .documentLink(Document.builder()
+                              .documentUrl("fake-url")
+                              .documentFileName("file-name")
+                              .documentBinaryUrl("binary-url")
+                              .build())
+            .build();
+    //Optional should be removed after refactoring
+    private final Optional<String> stitchStatus = Optional.of(randomAlphabetic(8));
+    private final Optional<Document> stitchedDocument = Optional.of(mock(Document.class));
```
Comment (filippo-possenti-version1) [inline src/test/java/uk/gov/hmcts/reform/civil/service/stitching/CivilDocumentStitchingServiceTest.java:62]: Can be local variables.

File: src/test/java/uk/gov/hmcts/reform/civil/service/stitching/CivilDocumentStitchingServiceTest.java [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+import static uk.gov.hmcts.reform.civil.model.documents.DocumentType.SEALED_CLAIM;
+import static uk.gov.hmcts.reform.civil.service.docmosis.DocmosisTemplates.N1;
+
+@ExtendWith(SpringExtension.class)
+@SpringBootTest(classes = {
+    CivilDocumentStitchingService.class,
+    JacksonAutoConfiguration.class,
+    CaseDetailsConverter.class
+})
+public class CivilDocumentStitchingServiceTest {
```
Comment (filippo-possenti-version1) [inline src/test/java/uk/gov/hmcts/reform/civil/service/stitching/CivilDocumentStitchingServiceTest.java:40]: The `public` modifier is not necessary in JUnit Jupiter. Remove it.

File: src/test/java/uk/gov/hmcts/reform/civil/service/stitching/CivilDocumentStitchingServiceTest.java [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+            )
+        );
+    }
+
+    public CaseData buildCaseData(Bundle bundle){
+        List<IdValue<Bundle>> caseBundles = List.of(new IdValue<>("1", bundle));
+        return CaseDataBuilder.builder().caseBundles(caseBundles).build();
+    }
+    @Test
+    public void whenCaseDataIsNullWithPayloadAndNoCaseBundlesNoStitchingDocument() {
```
Comment (filippo-possenti-version1) [inline src/test/java/uk/gov/hmcts/reform/civil/service/stitching/CivilDocumentStitchingServiceTest.java:79]: The `public` modifier is not necessary in JUnit Jupiter tests. Remove it from all the tests.

File: src/test/java/uk/gov/hmcts/reform/civil/service/stitching/CivilDocumentStitchingServiceTest.java [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+    }
+    @Test
+    public void whenCaseDataIsNullWithNoCaseBundles() {
+        List<DocumentMetaData> documentMetaDataList = buildDocumentMetaDataList();
+        CaseData caseData = CaseDataBuilder.builder().legacyCaseReference("ClaimNumber").build();
+        when(stitchingConfiguration.getStitchingUrl()).thenReturn("dummy_url");
+        CaseDocument caseDocument = civilDocumentStitchingService.bundle(documentMetaDataList, "Auth",
+                                                                         "Title", "FileName", caseData);
+        Assertions.assertNull(caseDocument);
+    }
```
Comment (filippo-possenti-version1) [inline src/test/java/uk/gov/hmcts/reform/civil/service/stitching/CivilDocumentStitchingServiceTest.java:94]: Put `// Given`, `// When`, `// Then` comments in all your tests to divide the three stages of each test. Better if along with them you clarify what's in each section. Example:
```
// Given: a case data with no case bundles
```

Chances are you'll also end up changing the `when(sticthing[...]` to a `given(stitching[...]`.

File: src/test/java/uk/gov/hmcts/reform/civil/service/stitching/CivilDocumentStitchingServiceTest.java [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+        when(stitchingConfiguration.getStitchingUrl()).thenReturn("dummy_url");
+        when(bundleRequestExecutor.post(any(BundleRequest.class), anyString(), anyString())).thenReturn(caseData1);
+        CaseDocument caseDocument = civilDocumentStitchingService.bundle(documentMetaDataList, "Auth",
+                                                                         "Title", "FileName", caseData);
+        Assertions.assertNull(caseDocument);
+    }
+    @Test
+    public void whenStitchingDocumentIsPresentAndCaseDataNotNullThenCaseDocumentNotNull() {
+        List<DocumentMetaData> documentMetaDataList = buildDocumentMetaDataList();
+        Bundle bundle = new Bundle(
```
Comment (filippo-possenti-version1) [inline src/test/java/uk/gov/hmcts/reform/civil/service/stitching/CivilDocumentStitchingServiceTest.java:109]: For consistency, use the builder here. It helps clarifying what fields you are setting.

**Review by filippo-possenti-version1 — CHANGES_REQUESTED** [review]
Some suggestions to improve overall readability. Marking as "Request changes" just to draw a high bar for this brand new test.

File: src/test/java/uk/gov/hmcts/reform/civil/service/stitching/CivilDocumentStitchingServiceTest.java (hunk 2 above)
Comment (UshaPanneerselvam1) [reply src/test/java/uk/gov/hmcts/reform/civil/service/stitching/CivilDocumentStitchingServiceTest.java:40]: Done

File: src/test/java/uk/gov/hmcts/reform/civil/service/stitching/CivilDocumentStitchingServiceTest.java (hunk 3 above)
Comment (UshaPanneerselvam1) [reply src/test/java/uk/gov/hmcts/reform/civil/service/stitching/CivilDocumentStitchingServiceTest.java:79]: Done

File: src/test/java/uk/gov/hmcts/reform/civil/service/stitching/CivilDocumentStitchingServiceTest.java (hunk 5 above)
Comment (UshaPanneerselvam1) [reply src/test/java/uk/gov/hmcts/reform/civil/service/stitching/CivilDocumentStitchingServiceTest.java:109]: Done

File: src/test/java/uk/gov/hmcts/reform/civil/service/stitching/CivilDocumentStitchingServiceTest.java [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+    }
+    @Test
+     void whenCaseDataIsNullWithPayloadAndNoCaseBundlesNoStitchingDocument() {
+        //Given: Payload details but case data not provided
+        List<DocumentMetaData> documentMetaDataList = new ArrayList<>();
+        CaseData caseData = CaseDataBuilder.builder().legacyCaseReference("ClaimNumber").build();
+        //Then: the case documents is null
+        CaseDocument caseDocument = civilDocumentStitchingService.bundle(documentMetaDataList, "Auth",
+                                                                  "Title", "FileName", caseData);
+        Assertions.assertNull(caseDocument);
```
Comment (filippo-possenti-version1) [inline src/test/java/uk/gov/hmcts/reform/civil/service/stitching/CivilDocumentStitchingServiceTest.java:80]: It's missing the `// When` section.

```suggestion
        //When: I try to bundle the document
        CaseDocument caseDocument = civilDocumentStitchingService.bundle(documentMetaDataList, "Auth",
                                                                  "Title", "FileName", caseData);
        //Then: the case documents is null
        Assertions.assertNull(caseDocument);
```

**Review by filippo-possenti-version1 — APPROVED** [review]

File: src/main/java/uk/gov/hmcts/reform/civil/service/stitching/CivilDocumentStitchingService.java [hunk 7]
```diff
@@ -43,7 +41,6 @@ public CaseDocument bundle(
         String bundleFilename,
         CaseData caseData
     ) {
-        CaseDocument caseDocument = null;
         CaseDetails payload =
```
Comment (filippo-possenti-version1) [inline src/main/java/uk/gov/hmcts/reform/civil/service/stitching/CivilDocumentStitchingService.java:44]: Minor: While we're at it, can I ask that this call be put on a single line? It's seems really overkill that they split it with each argument on a different line...

**Review by filippo-possenti-version1 — CHANGES_REQUESTED** [review]
Some changes for further simplification.

Also, be mindful with your commit messages. Your last commit suggests that you're doing unit-tests but here you already are at refactoring stage.

**Review by drummondjm — DISMISSED** [review]

File: src/main/java/uk/gov/hmcts/reform/civil/service/stitching/CivilDocumentStitchingService.java [hunk 8]
```diff
@@ -52,43 +49,46 @@ public CaseDocument bundle(
...
                 stitchingConfiguration.getStitchingUrl(),
                 authorisation
             );
         log.info("Called stitching api end point for {}", caseData.getLegacyCaseReference());
-        if (caseData1 != null) {
-            Optional<Document> stitchedDocument = caseData1.getCaseBundles().get(0).getValue().getStitchedDocument();
+        if(caseDataFromBundlePayload == null) {
+            log.info("Case data is null----------");
+            return null;
+        } else {
```
[the remark this answers is not in the document]
Comment (UshaPanneerselvam1) [reply src/main/java/uk/gov/hmcts/reform/civil/service/stitching/CivilDocumentStitchingService.java:62]: Done

File: src/main/java/uk/gov/hmcts/reform/civil/service/stitching/CivilDocumentStitchingService.java [hunk 9]
```diff
@@ -52,43 +49,46 @@ public CaseDocument bundle(
...
         }
-
-        return caseDocument;
     }
 
+    private CaseDocument retrieveCaseDocument(Optional<Document> stitchedDocument){
+        if(stitchedDocument.isEmpty()){
+            log.info("stitchedDocument is not present----------");
+            return null;
+        } else {
```
[the remark this answers is not in the document]
Comment (UshaPanneerselvam1) [reply src/main/java/uk/gov/hmcts/reform/civil/service/stitching/CivilDocumentStitchingService.java:75]: Done

File: src/main/java/uk/gov/hmcts/reform/civil/service/stitching/CivilDocumentStitchingService.java (hunk 7 above)
Comment (UshaPanneerselvam1) [reply src/main/java/uk/gov/hmcts/reform/civil/service/stitching/CivilDocumentStitchingService.java:44]: Done

**Review by filippo-possenti-version1 — APPROVED** [review]

**Review by filippo-possenti-version1 — DISMISSED** [review]

**Review by drummondjm — DISMISSED** [review]

**Review by filippo-possenti-version1 — APPROVED** [review]

**Pull request merged**